android/gltfio-android/src/main/cpp/ResourceLoader.cpp

```diff
@@ -52,12 +52,20 @@ Java_com_google_android_filament_gltfio_ResourceLoader_nAddResourceData(JNIEnv*
         jlong nativeLoader, jstring url, jobject javaBuffer, jint remaining) {
     ResourceLoader* loader = (ResourceLoader*) nativeLoader;
     AutoBuffer* buffer = new AutoBuffer(env, javaBuffer, remaining);
-    const char* curl = env->GetStringUTFChars(url, nullptr);
-    loader->addResourceData(curl,
+    const char* cstring = env->GetStringUTFChars(url, nullptr);
```

Comment:
Note: everywhere you call `GetStringUTFChars` you must also call `ReleaseStringUTFChars`. The JNI environment might have made a copy.

Reply:
Wow good catch, fixed the leak by amending the commit.